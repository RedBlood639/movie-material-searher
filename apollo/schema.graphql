enum Media {
  movie
  tv
}

type UpcomingResults {
  id: Int!
  media_type: Media!
  poster_path: String
  release_date: String!
  title: String!
}

type Upcoming {
  page: Int!
  results: [UpcomingResults!]!
  total_pages: Int!
}

type NowPlayingResults {
  id: Int!
  media_type: Media!
  poster_path: String
  title: String!
  vote_average: Float!
}

type NowPlaying {
  page: Int!
  results: [NowPlayingResults!]!
  total_pages: Int!
}

type MoviesSearchResults {
  id: Int!
  media_type: Media!
  title: String!
}

type MoviesSearch {
  cursor: Int
  hasMore: Boolean
  page: Int!
  results: [MoviesSearchResults!]!
  total_pages: Int!
}

type SimilarResults {
  id: Int!
  media_type: Media!
  poster_path: String
  release_date: String!
  title: String!
}

type SimilarMovies {
  cursor: Int
  hasMore: Boolean
  results: [SimilarResults!]!
}

type MovieInfo {
  backdrop_path: String
  budget: String!
  id: Int!
  media_type: Media!
  overview: String
  poster_path: String
  release_date: String!
  revenue: String!
  similar: SimilarMovies!
  title: String!
  vote_average: Float!
}

type TvShowInfo {
  backdrop_path: String
  id: Int!
  media_type: Media!
  number_of_episodes: Int!
  number_of_seasons: Int!
  overview: String
  poster_path: String
  release_date: String!
  similar: SimilarMovies!
  title: String!
  vote_average: Float!
}

type Query {
  nowPlaying(page: Int = 1): NowPlaying!
  movieInfo(id: ID!, cursor: Int, pageSize: Int = 4): MovieInfo!
  moviesSearch(query: String!, cursor: Int, pageSize: Int = 8): MoviesSearch!
  tvShowInfo(id: ID!, cursor: Int, pageSize: Int = 4): TvShowInfo!
  upcoming(page: Int = 1): Upcoming!
}
