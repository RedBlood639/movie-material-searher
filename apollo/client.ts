import {
  ApolloClient,
  InMemoryCache,
  makeVar,
  NormalizedCacheObject,
} from '@apollo/client'
import merge from 'deepmerge'
import isEqual from 'lodash.isequal'
import { useMemo } from 'react'

export type Favorite = {
  id: number
  media_type: string
  poster_path?: string | null
  title: string
}

export const favoritesVar = makeVar<Favorite[]>([])
export const prefersDarkModeVar = makeVar(false)

let apolloClient: ApolloClient<NormalizedCacheObject>

function createApolloClient() {
  return new ApolloClient({
    cache: new InMemoryCache(),
    ssrMode: typeof window === 'undefined',
    uri: process.env.NEXT_PUBLIC_SERVER_URL,
  })
}

export function initializeApollo(initialState: {} | null = null) {
  const internalApolloClient = apolloClient ?? createApolloClient()

  // If your page has Next.js data fetching methods that use Apollo Client, the initial state
  // gets hydrated here
  if (initialState) {
    // Get existing cache, loaded during client side data fetching
    const existingCache = internalApolloClient.extract()

    // Merge the existing cache into data passed from getStaticProps/getServerSideProps
    const data = merge(initialState, existingCache, {
      // combine arrays using object equality (like in sets)
      arrayMerge: (destinationArray, sourceArray) => [
        ...sourceArray,
        ...destinationArray.filter(d => sourceArray.every(s => !isEqual(d, s))),
      ],
    })

    // Restore the cache with the merged data
    internalApolloClient.cache.restore(data)
  }
  // For SSG and SSR always create a new Apollo Client
  if (typeof window === 'undefined') return internalApolloClient
  // Create the Apollo Client once in the client
  if (!apolloClient) apolloClient = internalApolloClient

  return internalApolloClient
}

const APOLLO_STATE_PROP_NAME = '__APOLLO_STATE__'

export function addApolloState(
  client: ApolloClient<NormalizedCacheObject>,
  pageProps: any,
) {
  if (pageProps?.props) {
    // eslint-disable-next-line no-param-reassign
    pageProps.props[APOLLO_STATE_PROP_NAME] = client.cache.extract()
  }

  return pageProps
}

export function useApollo(pageProps: any) {
  const state = pageProps[APOLLO_STATE_PROP_NAME]
  const store = useMemo(() => initializeApollo(state), [state])
  return store
}
