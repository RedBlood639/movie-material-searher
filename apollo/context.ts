import MoviesAPI from './datasource'

export type Context = {
  dataSources: {
    moviesAPI: MoviesAPI
  }
}
