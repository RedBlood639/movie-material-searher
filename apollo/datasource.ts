/* eslint-disable class-methods-use-this */
import { RequestOptions, RESTDataSource } from 'apollo-datasource-rest'
import dayjs from 'dayjs'
import numeral from 'numeral'

import {
  Media,
  MovieInfo,
  MoviesSearch,
  NowPlaying,
  TvShowInfo,
  Upcoming,
} from './__generated__'
import {
  MockMovieInfoResponse,
  MockMoviesSearchResponse,
  MockNowPlayingResponse,
  MockTvShowInfoResponse,
  MockUpcomingResponse,
} from './__responses__/raw-responses'

export default class MoviesAPI extends RESTDataSource {
  constructor() {
    super()
    this.baseURL = 'https://api.themoviedb.org/3'
  }

  protected willSendRequest(request: RequestOptions) {
    request.params.set('api_key', process.env.MOVIE_API_KEY as string)
  }

  private attachPoster(path: string, size = 200) {
    return path ? `https://image.tmdb.org/t/p/w${size}${path}` : null
  }

  private moviesUpcomingReducer(movies: MockUpcomingResponse): Upcoming {
    return {
      page: movies.page,
      results: movies.results.map(movie => ({
        id: movie.id,
        media_type: Media.Movie,
        poster_path: this.attachPoster(movie.poster_path),
        release_date: dayjs(movie.release_date).format('DD.MM.YYYY'),
        title: movie.title,
      })),
      total_pages: movies.total_pages,
    }
  }

  private moviesNowPlayingReducer(movies: MockNowPlayingResponse): NowPlaying {
    return {
      page: movies.page,
      results: movies.results.map(movie => ({
        id: movie.id,
        media_type: Media.Movie,
        poster_path: this.attachPoster(movie.poster_path),
        title: movie.title,
        vote_average: movie.vote_average,
      })),
      total_pages: movies.total_pages,
    }
  }

  private moviesSearchReducer(movies: MockMoviesSearchResponse): MoviesSearch {
    return {
      page: movies.page,
      results: movies.results
        .filter(movie => movie.media_type !== 'person')
        .map(movie => ({
          id: movie.id,
          media_type: movie.media_type as Media,
          title: movie?.title || movie?.name || '',
        })),
      total_pages: movies.total_pages,
    }
  }

  private movieInfoReducer(movie: MockMovieInfoResponse): MovieInfo {
    return {
      backdrop_path: this.attachPoster(movie.backdrop_path, 500),
      budget: numeral(movie.budget).format('$0,00'),
      id: movie.id,
      media_type: Media.Movie,
      overview: movie.overview,
      poster_path: this.attachPoster(movie.poster_path),
      release_date: movie.release_date
        ? dayjs(movie.release_date).format('DD MMMM YYYY')
        : '',
      revenue: numeral(movie.revenue).format('$0,00'),
      similar: {
        results: movie.similar.results.map(similarMovie => ({
          id: similarMovie.id,
          media_type: Media.Movie,
          poster_path: this.attachPoster(similarMovie.poster_path),
          release_date: similarMovie.release_date
            ? dayjs(similarMovie.release_date).format('YYYY')
            : '',
          title: similarMovie.title,
        })),
      },
      title: movie.title,
      vote_average: movie.vote_average,
    }
  }

  private tvShowInfoReducer(tv: MockTvShowInfoResponse): TvShowInfo {
    return {
      backdrop_path: this.attachPoster(tv.backdrop_path, 500),
      id: tv.id,
      media_type: Media.Tv,
      number_of_episodes: tv.number_of_episodes,
      number_of_seasons: tv.number_of_seasons,
      overview: tv.overview,
      poster_path: this.attachPoster(tv.poster_path),
      release_date: tv.first_air_date
        ? dayjs(tv.first_air_date).format('DD MMMM YYYY')
        : '',
      similar: {
        results: tv.similar.results.map(similarTvShow => ({
          id: similarTvShow.id,
          media_type: Media.Tv,
          poster_path: this.attachPoster(similarTvShow.poster_path),
          release_date: similarTvShow.first_air_date
            ? dayjs(similarTvShow.first_air_date).format('YYYY')
            : '',
          title: similarTvShow.name,
        })),
      },
      title: tv.name,
      vote_average: tv.vote_average,
    }
  }

  async getUpcoming(page: number) {
    const data: MockUpcomingResponse = await this.get('/movie/upcoming', {
      page,
    })
    return this.moviesUpcomingReducer(data)
  }

  async getNowPlaying(page: number) {
    const data: MockNowPlayingResponse = await this.get('/movie/now_playing', {
      page,
    })
    return this.moviesNowPlayingReducer(data)
  }

  async getMoviesSearch(query: string, page: number) {
    const data: MockMoviesSearchResponse = await this.get('/search/multi', {
      page,
      query,
    })
    return this.moviesSearchReducer(data)
  }

  async getMovieInfo(id: string) {
    const data: MockMovieInfoResponse = await this.get(`/movie/${id}`, {
      append_to_response: 'similar',
    })
    return this.movieInfoReducer(data)
  }

  async getTvShowInfo(id: string) {
    const data: MockTvShowInfoResponse = await this.get(`/tv/${id}`, {
      append_to_response: 'similar',
    })
    return this.tvShowInfoReducer(data)
  }
}
