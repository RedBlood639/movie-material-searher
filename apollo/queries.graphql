query GetMovieInfo($id: ID!, $pageSize: Int) {
  movieInfo(id: $id, pageSize: $pageSize) {
    backdrop_path
    budget
    id
    media_type
    overview
    poster_path
    release_date
    revenue
    similar {
      results {
        id
        media_type
        poster_path
        release_date
        title
      }
    }
    title
    vote_average
  }
}

query GetTvShowInfo($id: ID!, $pageSize: Int) {
  tvShowInfo(id: $id, pageSize: $pageSize) {
    backdrop_path
    id
    media_type
    number_of_episodes
    number_of_seasons
    overview
    poster_path
    release_date
    similar {
      results {
        id
        media_type
        poster_path
        release_date
        title
      }
    }
    title
    vote_average
  }
}

query GetMovies {
  nowPlaying {
    page
    results {
      id
      media_type
      poster_path
      title
      vote_average
    }
    total_pages
  }
  upcoming {
    page
    results {
      id
      media_type
      poster_path
      release_date
      title
    }
    total_pages
  }
}

query GetUpcoming($page: Int) {
  upcoming(page: $page) {
    page
    results {
      id
      media_type
      poster_path
      release_date
      title
    }
    total_pages
  }
}

query GetNowPlaying($page: Int) {
  nowPlaying(page: $page) {
    page
    results {
      media_type
      id
      poster_path
      title
      vote_average
    }
    total_pages
  }
}

query GetMoviesSearch($query: String!, $pageSize: Int) {
  moviesSearch(query: $query, pageSize: $pageSize) {
    results {
      id
      media_type
      title
    }
  }
}
