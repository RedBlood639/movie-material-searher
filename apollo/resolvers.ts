import { MoviesSearch, Resolvers, SimilarMovies } from './__generated__'

function paginateResults<T extends { results: { [key: string]: any } }>(
  data: T,
  size: number,
  cursor?: number | null,
) {
  let results: []

  if (!cursor) {
    results = data.results.slice(0, size)
  } else {
    results = data.results.slice(0, cursor + size)
  }

  return {
    ...data,
    cursor: results.length,
    hasMore: results.length !== data.results.length,
    results,
  }
}

const resolvers: Resolvers = {
  Query: {
    async movieInfo(_root, { cursor, id, pageSize }, { dataSources }) {
      try {
        const data = await dataSources.moviesAPI.getMovieInfo(id)
        return {
          ...data,
          similar: paginateResults<SimilarMovies>(
            data.similar,
            pageSize,
            cursor,
          ),
        }
      } catch (error) {
        throw new Error(error)
      }
    },
    async moviesSearch(_root, { cursor, pageSize, query }, { dataSources }) {
      const fetchMovies = async (page = 1): Promise<MoviesSearch> => {
        const data = await dataSources.moviesAPI.getMoviesSearch(query, page)

        if (!data.results.length && data.total_pages > 1) {
          return fetchMovies(data.page + 1)
        }

        return data
      }

      try {
        const data = await fetchMovies()
        return paginateResults<MoviesSearch>(data, pageSize, cursor)
      } catch (error) {
        throw new Error(error)
      }
    },
    async nowPlaying(_root, { page }, { dataSources }) {
      try {
        const data = await dataSources.moviesAPI.getNowPlaying(page)

        if (!data.results.length || data.page !== +page) {
          throw new Error('404 Not found')
        }

        return data
      } catch (error) {
        throw new Error(error)
      }
    },
    async tvShowInfo(_root, { cursor, id, pageSize }, { dataSources }) {
      try {
        const data = await dataSources.moviesAPI.getTvShowInfo(id)
        return {
          ...data,
          similar: paginateResults<SimilarMovies>(
            data.similar,
            pageSize,
            cursor,
          ),
        }
      } catch (error) {
        throw new Error(error)
      }
    },
    async upcoming(_root, { page }, { dataSources }) {
      try {
        const data = await dataSources.moviesAPI.getUpcoming(page)

        if (!data.results.length || data.page !== +page) {
          throw new Error('404 Not found')
        }

        return data
      } catch (error) {
        throw new Error(error)
      }
    },
  },
}

export default resolvers
