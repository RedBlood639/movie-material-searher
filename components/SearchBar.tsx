import { css } from '@emotion/react'
import styled from '@emotion/styled'
import { Autocomplete, TextField } from '@material-ui/core'
import { alpha } from '@material-ui/core/styles'
import {
  Movie as MovieIcon,
  Search as SearchIcon,
  Tv as TvIcon,
} from '@material-ui/icons'
import match from 'autosuggest-highlight/match'
import parse from 'autosuggest-highlight/parse'
import Link from 'next/link'
import { useState } from 'react'

import {
  MoviesSearchResults,
  useGetMoviesSearchLazyQuery,
} from '@/apollo/__generated__'

const StyledAutocomplete = styled(Autocomplete)`
  width: 70%;
  input {
    margin-left: ${({ theme }) => theme.spacing(5)};
  }
`
const StyledTextField = styled(TextField)`
  background-color: ${({ theme }) => alpha(theme.palette.common.black, 0.15)};
  border-radius: ${({ theme }) => theme.shape.borderRadius}px;
  ${({ theme }) => theme.breakpoints.up('sm')} {
    margin-left: ${({ theme }) => theme.spacing(1)};
  }
`
const StyledSearchIcon = styled(SearchIcon)`
  font-size: ${({ theme }) => theme.typography.pxToRem(25)};
  height: 100%;
  margin-left: ${({ theme }) => theme.spacing(2)};
  position: absolute;
  top: 0;
  user-select: none;
  ${({ theme }) => theme.breakpoints.up('sm')} {
    margin-left: ${({ theme }) => theme.spacing(3)};
  }
`
const TextHighlight = styled.span<{ highlight: boolean }>`
  ${props =>
    props.highlight &&
    css`
      font-weight: 700;
    `};
  white-space: pre-wrap;
`
const IconContainer = styled.span`
  padding-right: ${({ theme }) => theme.spacing(1)};
  vertical-align: text-top;
  svg {
    font-size: inherit;
  }
`

export default function SearchBar() {
  const [inputValue, setInputValue] = useState('')
  const [fetchMovies, { data, error, loading }] = useGetMoviesSearchLazyQuery()

  return (
    <StyledAutocomplete
      blurOnSelect
      disablePortal
      filterOptions={options => options}
      freeSolo
      getOptionLabel={(movie: any) => movie.title}
      id="autocomplete"
      inputValue={inputValue}
      loading={
        loading ||
        !!error ||
        (!data?.moviesSearch.results.length && !!inputValue.trim())
      }
      loadingText={loading ? 'Loading...' : error?.message || 'No results'}
      onInputChange={(_evt, value: string, reason) => {
        if (reason === 'input') {
          setInputValue(value)

          const newValue = value.trim()

          if (newValue && newValue !== inputValue.trim()) {
            fetchMovies({ variables: { query: newValue } })
          }
        }
        if (reason === 'clear') {
          setInputValue('')
        }
      }}
      openOnFocus
      options={(inputValue.trim() && data?.moviesSearch.results) || []}
      renderInput={params => (
        <>
          <StyledSearchIcon />
          <StyledTextField
            {...params}
            error={!!error}
            fullWidth
            id="search field"
            placeholder="Search..."
            size="small"
          />
        </>
      )}
      renderOption={(props, option: any, { inputValue: value }) => {
        const movie: MoviesSearchResults = option
        const matches = match(movie.title, value)
        const parts = parse(movie.title, matches)

        return (
          <Link href={`/${movie.media_type}/${movie.id}`}>
            <li {...props}>
              <span>
                <IconContainer>
                  {movie.media_type === 'movie' ? <MovieIcon /> : <TvIcon />}
                </IconContainer>
                {parts.map(part => (
                  <TextHighlight highlight={part.highlight}>
                    {part.text}
                  </TextHighlight>
                ))}
              </span>
            </li>
          </Link>
        )
      }}
    />
  )
}
