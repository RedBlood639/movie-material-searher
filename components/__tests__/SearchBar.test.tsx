import { GetMoviesSearchDocument } from '@/apollo/__generated__'
import { fireEvent, renderApollo, screen } from '@/lib/setupTests'

import SearchBar from '../SearchBar'

const mocks = [
  {
    request: {
      query: GetMoviesSearchDocument,
      variables: { query: 'title' },
    },
    result: {
      data: {
        moviesSearch: {
          results: [
            {
              id: 1,
              media_type: 'movie',
              title: 'movie title',
            },
            {
              id: 2,
              media_type: 'tv',
              title: 'tv title',
            },
          ],
        },
      },
    },
  },
]

describe('searchBar', () => {
  it('should check all cases', async () => {
    const { baseElement } = renderApollo(<SearchBar />, { mocks })

    const inputElement = screen.getByPlaceholderText('Search...')

    // open dropdown
    fireEvent.focus(inputElement)

    // check if empty value doesn't trigger request
    fireEvent.input(inputElement, { target: { value: '   ' } })

    // fetch movie
    fireEvent.input(inputElement, { target: { value: 'title' } })
    const movieSearchResultHighlight = await screen.findAllByText('title')
    const movieSearchResult = screen.getByText('movie')
    expect(movieSearchResultHighlight[0]).toBeInTheDocument()
    expect(movieSearchResult).toBeInTheDocument()
    expect(baseElement).toMatchSnapshot()

    // clear input on click
    const clearButton = screen.getByTitle('Clear')
    fireEvent.click(clearButton)
    expect(inputElement).toHaveProperty('value', '')
  })
})
