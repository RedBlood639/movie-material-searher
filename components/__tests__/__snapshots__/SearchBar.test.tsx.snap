// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchBar should check all cases 1`] = `
<body>
  <div>
    <div
      aria-expanded="true"
      aria-owns="autocomplete-listbox"
      class="MuiAutocomplete-root Mui-focused MuiAutocomplete-hasClearIcon ep0uteb4 css-78p48u-MuiAutocomplete-root-StyledAutocomplete"
      role="combobox"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium ep0uteb2 css-1cntsc7-MuiSvgIcon-root-StyledSearchIcon"
        data-testid="SearchIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
      <div
        class="MuiFormControl-root MuiFormControl-marginNone MuiFormControl-fullWidth MuiTextField-root ep0uteb3 css-1oyzkgj-MuiFormControl-root-StyledTextField"
      >
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-4a33cb-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-autocomplete="list"
            aria-controls="autocomplete-listbox"
            aria-invalid="false"
            autocapitalize="none"
            autocomplete="off"
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused MuiAutocomplete-input MuiAutocomplete-inputFocused css-b55ri7-MuiInputBase-input-MuiOutlinedInput-input"
            id="autocomplete"
            placeholder="Search..."
            spellcheck="false"
            type="text"
          />
          <div
            class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
          >
            <button
              aria-label="Clear"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-clearIndicator css-1ou7wa2-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-clearIndicator"
              tabindex="-1"
              title="Clear"
              type="button"
            >
              <span
                class="MuiIconButton-label css-4jkopv-MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                  data-testid="CloseIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1o2jng6-MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="css-1br1jja"
              style="width: 0.01px;"
            >
              <span
                class="notranslate"
              >
                â€‹
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="MuiAutocomplete-popper MuiAutocomplete-popperDisablePortal css-p8t6er-MuiAutocomplete-popper"
      data-popper-escaped=""
      data-popper-placement="bottom"
      data-popper-reference-hidden=""
      role="presentation"
      style="position: absolute; top: 0px; left: 0px; width: 0px; margin: 0px; transform: translate(0px, 0px);"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAutocomplete-paper css-o3wlso-MuiPaper-root-MuiAutocomplete-paper"
      >
        <ul
          aria-labelledby="autocomplete-label"
          class="MuiAutocomplete-listbox css-1vfr3ul-MuiAutocomplete-listbox"
          id="autocomplete-listbox"
          role="listbox"
        >
          <li
            aria-disabled="false"
            aria-selected="false"
            class="MuiAutocomplete-option"
            data-option-index="0"
            id="autocomplete-option-0"
            role="option"
            tabindex="-1"
          >
            <span>
              <span
                class="css-1n2wxre-IconContainer ep0uteb0"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="MovieIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18 4l2 4h-3l-2-4h-2l2 4h-3l-2-4H8l2 4H7L5 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V4h-4z"
                  />
                </svg>
              </span>
              <span
                class="css-ztfl0n-TextHighlight ep0uteb1"
              >
                movie 
              </span>
              <span
                class="css-1gbrnq5-TextHighlight-TextHighlight ep0uteb1"
              >
                title
              </span>
            </span>
          </li>
          <li
            aria-disabled="false"
            aria-selected="false"
            class="MuiAutocomplete-option"
            data-option-index="1"
            id="autocomplete-option-1"
            role="option"
            tabindex="-1"
          >
            <span>
              <span
                class="css-1n2wxre-IconContainer ep0uteb0"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="TvIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M21 3H3c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h5v2h8v-2h5c1.1 0 1.99-.9 1.99-2L23 5c0-1.1-.9-2-2-2zm0 14H3V5h18v12z"
                  />
                </svg>
              </span>
              <span
                class="css-ztfl0n-TextHighlight ep0uteb1"
              >
                tv 
              </span>
              <span
                class="css-1gbrnq5-TextHighlight-TextHighlight ep0uteb1"
              >
                title
              </span>
            </span>
          </li>
        </ul>
      </div>
    </div>
  </div>
</body>
`;
